caching-service/src/main/java/org/zowe/apiml/caching/service/infinispan/storage/InfinispanStorage.java

```diff
@@ -153,4 +151,75 @@ public void deleteForService(String serviceId) {
             }
         });
     }
+
+    @Override
+    public void deleteItemFromMap(String serviceId, String mapKey) {
+        if (mapKey.equals("invalidUsers") || mapKey.equals("invalidScopes")) {
+            removeNonRelevantRules(serviceId, mapKey);
+        } else {
+            removeNonRelevantTokens(serviceId, mapKey);
+        }
+    }
+
+    private void removeNonRelevantTokens(String serviceId, String mapKey) {
+        Map<String, String> map = tokenCache.get(serviceId + mapKey);
+        if (map != null && !map.isEmpty()) {
+            LocalDateTime timestamp = LocalDateTime.now();
+            ConcurrentMap<String,String> concurrentMap = new ConcurrentHashMap<>(map);
```

Comment:
no need to use concurrentmap for local variable

Reply:
if I don't use, I get an `ConcurrentModificationException`